File: flutter_frontend/lib/models/post.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:universe/models/user.dart';

class Post {
  User? user;
  String? id;
  final String title;
  final String body;
  final String authorId;
  final List<dynamic> images;
  final List<dynamic> videos;
  final DateTime publishDate;

  Post({
    this.id,
    required this.title,
    required this.body,
    required this.authorId,
    required this.images,
    required this.videos,
    required this.publishDate,
  });

  factory Post.fromFirestore(
    DocumentSnapshot<Map<String, dynamic>> snapshot,
    SnapshotOptions? options,
  ) {
    final data = snapshot.data();
    return Post(
      id: snapshot.reference.path,
      title: data?['title'],
      body: data?['body'],
      authorId: data?['author'],
      images: data?['images'],
      videos: data?['videos'],
      publishDate: data?['publishDate'].toDate(),
    );
  }

  Map<String, dynamic> toFirestore() {
    return {
      "title": title,
      "body": body,
      "author": authorId,
      "images": images,
      "videos": videos,
      "publishDate": Timestamp.fromDate(publishDate),
    };
  }
}

File: Universe_Backend/Data/Post.cs
namespace UniverseBackend.Data;

public class Post
{
    public required int ID { get; set; }
    public required int AuthorID { get; set; }
    public required string Title { get; set; }
    public required string Body { get; set; }
    public required List<string> Images { get; set; }
    public required List<string> Videos { get; set; }
    public required DateTime PublishDate { get; set; }
    public required int ParentPostID { get; set; }
}